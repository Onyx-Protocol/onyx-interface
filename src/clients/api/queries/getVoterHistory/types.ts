import { ProposalApiResponse } from '../getProposals';
import { VoterResult } from '../getVoters';

export interface GetVoterHistoryResponse {
  data: {
    voter: VoterResult;
    proposal: ProposalApiResponse;
  }[];
  metadata: {
    page: number;
    limit: number;
    totalItem: number;
    totalPage: number;
  };
}
