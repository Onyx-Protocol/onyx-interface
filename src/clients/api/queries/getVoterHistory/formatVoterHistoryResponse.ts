import BigNumber from 'bignumber.js';
import { formatToProposal } from 'utilities';

import indexedVotingSupportNames from 'constants/indexedVotingSupportNames';

import { getGovernorBravoDelegateContract } from '../../../contracts';
import getWeb3NoAccount from '../../../web3/getWeb3NoAccount';
import { GetVoterHistoryResponse } from './types';

const formatVoterHistoryResponse = async (
  data: GetVoterHistoryResponse,
  { limit, page }: { limit: number; page: number },
) => {
  const web3NoAccount = getWeb3NoAccount();
  const latestBlock = await web3NoAccount.eth.getBlock('latest');
  const governorBravoDelegateContract = getGovernorBravoDelegateContract(web3NoAccount);
  const quorumVotes = await governorBravoDelegateContract.methods.quorumVotes().call();

  return {
    limit,
    page,
    total: data.length < limit ? page * limit + data.length : (page + 1) * limit + 1,
    voterHistory: await Promise.all(
      data.map(async d => ({
        address: d.address,
        blockNumber: +d.blockNumber,
        blockTimestamp: +d.blockTimestamp,
        createdAt: new Date(+d.blockTimestamp * 1000),
        id: d.id,
        proposal: await formatToProposal(d.proposal, new BigNumber(quorumVotes), {
          latestBlockTimestamp: Number(latestBlock.timestamp),
          latestBlockNumber: latestBlock.number,
        }),
        reason: undefined,
        support: indexedVotingSupportNames[Number(d.support)],
        updatedAt: new Date(+d.blockTimestamp * 1000),
        votesWei: new BigNumber(d.votes),
      })),
    ),
  };
};

export default formatVoterHistoryResponse;
