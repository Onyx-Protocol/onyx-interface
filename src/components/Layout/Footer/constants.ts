export const XCN_MEDIUM_URL = 'https://medium.com/OnyxProtocol';
export const XCN_DISCORD_URL = 'https://discord.com/invite/onyxprotocol';
export const XCN_TELEGRAM_URL = 'https://t.me/OnyxOrg';
export const XCN_TWITTER_URL = 'https://twitter.com/OnyxProtocol';
export const XCN_GITHUB_URL = 'https://GitHub.com/Onyx-Protocol';
