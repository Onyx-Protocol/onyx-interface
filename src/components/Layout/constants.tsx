import Path from 'constants/path';

import { MenuItem } from './types';

/**
 * Updated by ozdav
 * Just commented to hide some menu links
 */
export const menuItems: MenuItem[] = [
  {
    href: Path.ROOT,
    // Translation key: do not remove this comment
    // t('layout.menuItems.staking')
    i18nKey: 'layout.menuItems.stake',
    i18nTitleKey: 'layout.menuItems.stake',
    icon: 'stake',
  },
  // {
  //   href: Path.ROOT,
  //   // Translation key: do not remove this comment
  //   // t('layout.menuItems.dashboard')
  //   i18nKey: 'layout.menuItems.dashboard',
  //   i18nTitleKey: 'layout.menuItems.dashboard',
  //   icon: 'dashboard',
  // },
  // {
  //   href: Path.MARKETS,
  //   // Translation key: do not remove this comment
  //   // t('layout.menuItems.markets')
  //   i18nKey: 'layout.menuItems.markets',
  //   i18nTitleKey: 'layout.menuItems.markets',
  //   icon: 'market',
  // },
  {
    href: Path.HISTORY,
    // Translation key: do not remove this comment
    // t('layout.menuItems.history')
    i18nKey: 'layout.menuItems.history',
    i18nTitleKey: 'layout.menuItems.history',
    icon: 'history',
  },
  {
    href: Path.SWAP,
    // Translation key: do not remove this comment
    // t('layout.menuItems.swap')
    i18nKey: 'layout.menuItems.swap',
    i18nTitleKey: 'layout.menuItems.swap',
    icon: 'convert',
    // isNew: true,
  },
  {
    href: Path.FARM,
    // Translation key: do not remove this comment
    // t('layout.menuItems.farm')
    i18nKey: 'layout.menuItems.farm',
    i18nTitleKey: 'layout.menuItems.farm',
    icon: 'farm',
    // isNew: true,
  },
  {
    href: Path.GOVERNANCE,
    // Translation key: do not remove this comment
    // t('layout.menuItems.governance')
    i18nKey: 'layout.menuItems.governance',
    i18nTitleKey: 'layout.menuItems.governance',
    icon: 'vote',
  },
  {
    href: Path.AIAGENT,
    // Translation key: do not remove this comment
    // t('layout.menuItems.governance')
    i18nKey: 'layout.menuItems.aiagent',
    i18nTitleKey: 'layout.menuItems.aiagent',
    icon: 'aiagent',
    isNew: true,
  },
  // {
  //   href: Path.XCN,
  //   // Translation key: do not remove this comment
  //   // t('layout.menuItems.xcn')
  //   i18nKey: 'layout.menuItems.xcn',
  //   i18nTitleKey: 'layout.menuItems.xcn',
  //   icon: 'xcnOutlined',
  // },
  // {
  //   href: Path.LIQUIDATE,
  //   // Translation key: do not remove this comment
  //   // t('layout.menuItems.xcn')
  //   i18nKey: 'layout.menuItems.liquidate',
  //   i18nTitleKey: 'layout.menuItems.liquidate',
  //   icon: 'liquidate',
  // },
  // {
  //   href: Path.WPUNKS,
  //   // Translation key: do not remove this comment
  //   // t('layout.menuItems.xcn')
  //   i18nKey: 'layout.menuItems.fpunk',
  //   i18nTitleKey: 'layout.menuItems.fpunk',
  //   icon: 'wpunksOutlined',
  // },
  // TO BE ACTIVATED ONCE CALCULATOR IS READY
  {
    href: Path.GOVERNANCE_LEADER_BOARD,
    // Translation key: do not remove this comment
    // t('layout.menuItems.xcn')
    i18nKey: 'layout.menuItems.leaderboard',
    i18nTitleKey: 'layout.menuItems.leaderboard',
    isNew: true,
    icon: 'xcnOutlined',
  },
  {
    href: Path.POINTS,
    // Translation key: do not remove this comment
    // t('layout.menuItems.points')
    i18nKey: 'layout.menuItems.points',
    i18nTitleKey: 'layout.menuItems.points',
    icon: 'gem',
    isNew: true,
    subMenu: [
      {
        href: Path.POINTS,
        // Translation key: do not remove this comment
        // t('layout.menuItems.pointsDashboard')
        i18nKey: 'layout.menuItems.pointsDashboard',
        i18nTitleKey: 'layout.menuItems.pointsDashboard',
        icon: 'dashboard',
      },
      {
        href: Path.POINTS_LEADERBOARD,
        // Translation key: do not remove this comment
        // t('layout.menuItems.pointsLeaderboard')
        i18nKey: 'layout.menuItems.pointsLeaderboard',
        i18nTitleKey: 'layout.menuItems.pointsLeaderboard',
        icon: 'person',
      },
    ],
  },
];
