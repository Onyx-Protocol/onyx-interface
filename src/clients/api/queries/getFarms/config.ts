import { EthChainId } from 'types';

import { TOKENS, UNISWAP_TOKENS } from 'constants/tokens';

import { FarmConfig } from './types';

const farmConfig: FarmConfig[] = [
  {
    pid: 0,
    lpSymbol: 'XCN-WETH LP',
    lpAddresses: {
      [EthChainId.MAINNET]: '0x859f7092f56c43BB48bb46dE7119d9c799716CDF',
      [EthChainId.TESTNET]: '',
    },
    token: TOKENS.xcn,
    quoteToken: UNISWAP_TOKENS.weth,
  },

  // {
  //   pid: 2,
  //   lpSymbol: 'XCN-WETH LP',
  //   lpAddresses: {
  //     [EthChainId.MAINNET]: '',
  //     [EthChainId.TESTNET]: '0x006Df670C4735A18422B8c75F06d891D67e6B405',
  //   },
  //   token: TOKENS.xcn,
  //   quoteToken: UNISWAP_TOKENS.weth,
  // },

  // {
  //   pid: 3,
  //   lpSymbol: 'XCN-USDT LP',
  //   lpAddresses: {
  //     [EthChainId.MAINNET]: '',
  //     [EthChainId.TESTNET]: '0xe346BcA6aDFCA5E6474a43E1d73270E43e80F41e',
  //   },
  //   token: TOKENS.xcn,
  //   quoteToken: TOKENS.usdt,
  // },
];

export default farmConfig;
