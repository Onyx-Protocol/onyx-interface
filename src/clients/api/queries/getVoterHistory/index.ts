import config from 'config';
import { VoterHistory } from 'types';

import getVoterHistorySubGraph from 'utilities/getVoterHistorySubGraph';

import formatVoterHistoryResponse from './formatVoterHistoryResponse';

export interface GetVoterHistoryInput {
  page?: number;
  address: string;
}

export interface GetVoterHistoryOutput {
  voterHistory: VoterHistory[];
  limit: number;
  page: number;
  total: number;
}

const getVoterHistory = async ({
  page = 0,
  address,
}: GetVoterHistoryInput): Promise<GetVoterHistoryOutput> => {
  const proposalVotes = await getVoterHistorySubGraph(config.chainId, address, {
    limit: 5,
    offset: page * 5,
  });

  return formatVoterHistoryResponse(proposalVotes, { page, limit: 5 });
};

export default getVoterHistory;
