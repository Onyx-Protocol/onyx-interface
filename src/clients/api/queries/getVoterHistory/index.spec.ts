import config from 'config';
import { VError } from 'errors';
import { enableFetchMocks } from 'jest-fetch-mock';

import voterHistoryResponse from '__mocks__/api/voterHistory.json';
import fakeAddress from '__mocks__/models/address';
import { SUBGRAPH_LINKS } from 'constants/endpoints';

import getVoterHistory from '.';

enableFetchMocks();

describe('api/queries/getVoterHistory', () => {
  test('returns formatted voter history', async () => {
    fetchMock.mockResponseOnce(
      JSON.stringify({
        status: 200,
        data: {
          proposalVotes: voterHistoryResponse,
        },
      }),
    );

    const payload = await getVoterHistory({
      page: 2,
      address: fakeAddress,
    });

    expect(fetchMock).toBeCalledWith(SUBGRAPH_LINKS[config.chainId].latest, {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
      },
      body: JSON.stringify({
        query: `
          query proposalVotesQuery {
            proposalVotes(
              first: 5,
              skip: 10,
              where: { address: "${fakeAddress}" }
              orderBy: proposal__createdBlockTimestamp,
              orderDirection: desc
            ) {
              id
              proposal {
                id
                proposer
                targets
                values
                signatures
                callDatas
                startBlock
                endBlock
                description
                state
                eta
                forVotes
                againstVotes
                createdBlockNumber
                createdBlockTimestamp
                createdTransactionHash
                queuedBlockNumber
                queuedBlockTimestamp
                queuedTransactionHash
                executedBlockNumber
                executedBlockTimestamp
                executedTransactionHash
                canceledBlockNumber
                canceledBlockTimestamp
                canceledTransactionHash
              }
              address
              support
              votes
              blockNumber
              blockTimestamp
              transactionHash
            }
          }
        `,
      }),
    });

    expect(payload).toMatchSnapshot();
  });

  test('Gets called with correct default arguments', async () => {
    fetchMock.mockResponseOnce(
      JSON.stringify({
        status: 200,
        data: {
          proposalVotes: voterHistoryResponse,
        },
      }),
    );

    const payload = await getVoterHistory({ address: fakeAddress });

    expect(fetchMock).toBeCalledWith(SUBGRAPH_LINKS[config.chainId].latest, {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
      },
      body: JSON.stringify({
        query: `
          query proposalVotesQuery {
            proposalVotes(
              first: 5,
              skip: 0,
              where: { address: "${fakeAddress}" }
              orderBy: proposal__createdBlockTimestamp,
              orderDirection: desc
            ) {
              id
              proposal {
                id
                proposer
                targets
                values
                signatures
                callDatas
                startBlock
                endBlock
                description
                state
                eta
                forVotes
                againstVotes
                createdBlockNumber
                createdBlockTimestamp
                createdTransactionHash
                queuedBlockNumber
                queuedBlockTimestamp
                queuedTransactionHash
                executedBlockNumber
                executedBlockTimestamp
                executedTransactionHash
                canceledBlockNumber
                canceledBlockTimestamp
                canceledTransactionHash
              }
              address
              support
              votes
              blockNumber
              blockTimestamp
              transactionHash
            }
          }
        `,
      }),
    });

    expect(payload.voterHistory).toHaveLength(3);

    expect(payload).toMatchSnapshot();
  });
});
