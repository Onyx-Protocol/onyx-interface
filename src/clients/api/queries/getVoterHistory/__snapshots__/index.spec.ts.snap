// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`api/queries/getVoterHistory Gets called with correct default arguments 1`] = `
Object {
  "limit": 5,
  "page": 0,
  "total": 3,
  "voterHistory": Array [
    Object {
      "address": "0x9d5be40cf72707d07513a3915afa8a3ad84f31e9",
      "blockNumber": 17762676,
      "blockTimestamp": 1690198247,
      "createdAt": 2023-07-24T11:30:47.000Z,
      "id": "0x006ac40ad30b592d6f21945e5e297f50720ad9cfa100642244647d4ea6ba20ceab000000",
      "proposal": Object {
        "actions": Array [
          Object {
            "callData": "0x0000000000000000000000004410cbf371a86355ce0690fd08a92373dcd8e57f00000000000000000000000000000000000000000168a6975c3504b634000000",
            "signature": "grantDAO(address,uint256)",
            "target": "0x28ca9caae31602d0312ebf6466c9dd57fca5da93",
            "value": "0",
          },
        ],
        "againstVotesWei": "0",
        "cancelDate": undefined,
        "cancelTxHash": undefined,
        "createdDate": 2023-07-23T17:22:11.000Z,
        "createdTxHash": "0x241eade249b1dc53693ad371d36c824313d97bb754d888899d146d25358a3fb8",
        "description": Object {
          "description": "The Onyx DAO proposes to allocate a budget for current operations in Augâ€“Oct 2023:

- Development and Infrastructure support
- Marketing and Management

Total is 436,000,000 XCN.

Details and discussion are in the community forum: https://community.onyx.org/t/budget-allocation-for-current-operations-august-october-2023/768 ",
          "title": "OIP-19: Budget allocation for current operations",
          "version": "v2",
        },
        "endBlock": 17770072,
        "executedDate": 2023-07-27T17:47:35.000Z,
        "executedTxHash": "0x1dcc4466b73db4481ee1969e15799a1608beddde06bceb596e27f1d527b614de",
        "forVotesWei": "2.35266819114955989350637682e+26",
        "id": 19,
        "isEnded": false,
        "isStarted": false,
        "proposer": "0xb0f1ff21e71f5e5fe1d98309e6c6595bff625615",
        "queuedDate": 2023-07-25T17:42:47.000Z,
        "queuedTxHash": "0xf5a2b0d62e48e725697ebfcde5613744f0bd22b3506b1bea1819037982524e6c",
        "startBlock": 17757272,
        "state": "Pending",
        "totalVotesWei": "2.35266819114955989350637682e+26",
      },
      "reason": undefined,
      "support": "FOR",
      "updatedAt": 2023-07-24T11:30:47.000Z,
      "votesWei": "0",
    },
    Object {
      "address": "0xbabbf5261afbdb1f1353b574f5cae76ae916d315",
      "blockNumber": 16899221,
      "blockTimestamp": 1679683451,
      "createdAt": 2023-03-24T18:44:11.000Z,
      "id": "0x00db9beebc743dbcdcd34c1ac647bd4bb8f2c2ee3506d6f338921f1cba0c91d38c000000",
      "proposal": Object {
        "actions": Array [
          Object {
            "callData": "0x",
            "signature": "_acceptAdmin()",
            "target": "0x7d61ed92a6778f5abf5c94085739f1edabec2800",
            "value": "0",
          },
          Object {
            "callData": "0x",
            "signature": "_acceptAdmin()",
            "target": "0x1961ad247b47f4f2242e55a0e5578c6cf01f8d12",
            "value": "0",
          },
          Object {
            "callData": "0x",
            "signature": "_acceptAdmin()",
            "target": "0x714bd93ab6ab2f0bcfd2aeaf46a46719991d0d79",
            "value": "0",
          },
          Object {
            "callData": "0x",
            "signature": "_acceptAdmin()",
            "target": "0x1933f1183c421d44d531ed40a5d2445f6a91646d",
            "value": "0",
          },
          Object {
            "callData": "0x",
            "signature": "_acceptAdmin()",
            "target": "0xbced4e924f28f43a24ceedec69ee21ed4d04d2dd",
            "value": "0",
          },
          Object {
            "callData": "0x",
            "signature": "_acceptAdmin()",
            "target": "0x8f35113cfaba700ed7a907d92b114b44421e412a",
            "value": "0",
          },
          Object {
            "callData": "0x",
            "signature": "_acceptAdmin()",
            "target": "0x830dacd5d0a62afa92c9bc6878461e9cd317b085",
            "value": "0",
          },
          Object {
            "callData": "0x",
            "signature": "_acceptAdmin()",
            "target": "0xfee4428b7f403499c50a6da947916b71d33142dc",
            "value": "0",
          },
          Object {
            "callData": "0x",
            "signature": "_acceptAdmin()",
            "target": "0x7497e945f23192d1dc3095409837f67927170f2d",
            "value": "0",
          },
          Object {
            "callData": "0x",
            "signature": "_acceptAdmin()",
            "target": "0xc0534bb221c0963b213f417585e3ca581993793a",
            "value": "0",
          },
          Object {
            "callData": "0x",
            "signature": "_acceptAdmin()",
            "target": "0xf52459e8bc708d55400c167e0d77ed49641226d3",
            "value": "0",
          },
          Object {
            "callData": "0x",
            "signature": "_acceptAdmin()",
            "target": "0x307269cf921213bd2e16b22dcd4401aec21d00bf",
            "value": "0",
          },
          Object {
            "callData": "0x",
            "signature": "_acceptAdmin()",
            "target": "0x11b258a1c5ec962f43ab523be9b8fa317cb2eb69",
            "value": "0",
          },
          Object {
            "callData": "0x",
            "signature": "_acceptAdmin()",
            "target": "0x4476175752a4947de953e26b40fc8c1e145fea82",
            "value": "0",
          },
          Object {
            "callData": "0x",
            "signature": "_acceptAdmin()",
            "target": "0x0c19d213e9f2a5cbaa4ec6e8eac55a22276b0641",
            "value": "0",
          },
          Object {
            "callData": "0x",
            "signature": "_acceptAdmin()",
            "target": "0x3f2dd7adbcd464d5b0a766f7eb367bd20a249691",
            "value": "0",
          },
          Object {
            "callData": "0x",
            "signature": "_acceptAdmin()",
            "target": "0x7ab16ed27ab8b22cb625db55c0e58521969cb091",
            "value": "0",
          },
        ],
        "againstVotesWei": "0",
        "cancelDate": undefined,
        "cancelTxHash": undefined,
        "createdDate": 2023-03-24T12:13:11.000Z,
        "createdTxHash": "0xdfebb830682e5a776850d057eb908636558c0d49d8a6c1020010a76cf76db5cd",
        "description": Object {
          "description": "Abstract: After 3 weeks of testing and live network activity, the Onyx Liquidity Protocol (OLP) will now be fully shifted to the Onyx Governance and Onyx DAO . This proposal is deployed in two parts which will migrate the ownership of the OLP smart contracts to the time lock which is controlled by the Onyx Governance Onyxcoin $XCN stakers who secure and govern the network. Alongside this proposal, the OLP is now fully available via IPFS, a decentralized hosting making Onyx and OLP fully decentralized from the smart contract level to hosting.
XCN holders are asked to vote in favor for this proposal to transfer OLP to Onyx DAO and Governance.",
          "title": "OIP-9: Launch OLP Governance",
          "version": "v2",
        },
        "endBlock": 16910087,
        "executedDate": 2023-03-28T17:37:23.000Z,
        "executedTxHash": "0xddbe09d6597b72d88f93f9f31ec630c01a1c384cae9ab454a562f113ba834e0c",
        "forVotesWei": "2.873025810619257470619105486e+27",
        "id": 8,
        "isEnded": false,
        "isStarted": false,
        "proposer": "0xa8738f54c6c31ea559565a468056df233e6f4c12",
        "queuedDate": 2023-03-26T16:37:47.000Z,
        "queuedTxHash": "0x7d1960521610acfe101e477ca7a39ae8b2296f97097373347c8f196f050657a8",
        "startBlock": 16897287,
        "state": "Pending",
        "totalVotesWei": "2.873025810619257470619105486e+27",
      },
      "reason": undefined,
      "support": "FOR",
      "updatedAt": 2023-03-24T18:44:11.000Z,
      "votesWei": "0",
    },
    Object {
      "address": "0x7cb436777bcff797f465a96f6b34383e10a43483",
      "blockNumber": 16743192,
      "blockTimestamp": 1677788351,
      "createdAt": 2023-03-02T20:19:11.000Z,
      "id": "0x040e1a4716ce3f28c60835d4f6cdd8c3b6ca3e9d42e08b0d58792519aefe063942000000",
      "proposal": Object {
        "actions": Array [
          Object {
            "callData": "0x00000000000000000000000000000000000000000000002a5a058fc295ed0000",
            "signature": "setRewardPerblock(uint256)",
            "target": "0x23445c63feef8d85956dc0f19ade87606d0e19a9",
            "value": "0",
          },
        ],
        "againstVotesWei": "0",
        "cancelDate": undefined,
        "cancelTxHash": undefined,
        "createdDate": 2023-03-02T20:09:47.000Z,
        "createdTxHash": "0x40e386129ea53babe8084dc68449cb711cdb7661c2b8344ca1d15ff0d48c6f20",
        "description": Object {
          "description": "## Increase Staking Rewards

Onyx DAO proposes to the community to increase the staking vault rewards which would make the APR approximately 33% as opposed to todays approximately 6.7% staking rate.

By making this change the staking rewards still remain under the constitution threshold and creates incentive rewards for those who are securing and voting on the Onyx Protocol and Onyx DAO.

Discussion: https://community.onyx.org/t/oip-8-increase-staking-rewards/35",
          "title": "OIP-8: Increase Staking Rewards",
          "version": "v2",
        },
        "endBlock": 16755946,
        "executedDate": 2023-03-07T20:39:35.000Z,
        "executedTxHash": "0x857d533ae969d5e26fea14447bf8d5f0c81bbef97daaee4078148e626dfea6f8",
        "forVotesWei": "3.02286978509899863154493852e+26",
        "id": 7,
        "isEnded": false,
        "isStarted": false,
        "proposer": "0x09a74d465c49fd07d6eaf2028c7cff31c93d3a74",
        "queuedDate": 2023-03-05T16:31:35.000Z,
        "queuedTxHash": "0xb804ff32c580407f7d45a443de08029a6bf05670c480d324d2bca5c6e4761a7a",
        "startBlock": 16743146,
        "state": "Pending",
        "totalVotesWei": "3.02286978509899863154493852e+26",
      },
      "reason": undefined,
      "support": "FOR",
      "updatedAt": 2023-03-02T20:19:11.000Z,
      "votesWei": "2.04949456e+24",
    },
  ],
}
`;

exports[`api/queries/getVoterHistory returns formatted voter history 1`] = `
Object {
  "limit": 5,
  "page": 2,
  "total": 13,
  "voterHistory": Array [
    Object {
      "address": "0x9d5be40cf72707d07513a3915afa8a3ad84f31e9",
      "blockNumber": 17762676,
      "blockTimestamp": 1690198247,
      "createdAt": 2023-07-24T11:30:47.000Z,
      "id": "0x006ac40ad30b592d6f21945e5e297f50720ad9cfa100642244647d4ea6ba20ceab000000",
      "proposal": Object {
        "actions": Array [
          Object {
            "callData": "0x0000000000000000000000004410cbf371a86355ce0690fd08a92373dcd8e57f00000000000000000000000000000000000000000168a6975c3504b634000000",
            "signature": "grantDAO(address,uint256)",
            "target": "0x28ca9caae31602d0312ebf6466c9dd57fca5da93",
            "value": "0",
          },
        ],
        "againstVotesWei": "0",
        "cancelDate": undefined,
        "cancelTxHash": undefined,
        "createdDate": 2023-07-23T17:22:11.000Z,
        "createdTxHash": "0x241eade249b1dc53693ad371d36c824313d97bb754d888899d146d25358a3fb8",
        "description": Object {
          "description": "The Onyx DAO proposes to allocate a budget for current operations in Augâ€“Oct 2023:

- Development and Infrastructure support
- Marketing and Management

Total is 436,000,000 XCN.

Details and discussion are in the community forum: https://community.onyx.org/t/budget-allocation-for-current-operations-august-october-2023/768 ",
          "title": "OIP-19: Budget allocation for current operations",
          "version": "v2",
        },
        "endBlock": 17770072,
        "executedDate": 2023-07-27T17:47:35.000Z,
        "executedTxHash": "0x1dcc4466b73db4481ee1969e15799a1608beddde06bceb596e27f1d527b614de",
        "forVotesWei": "2.35266819114955989350637682e+26",
        "id": 19,
        "isEnded": false,
        "isStarted": false,
        "proposer": "0xb0f1ff21e71f5e5fe1d98309e6c6595bff625615",
        "queuedDate": 2023-07-25T17:42:47.000Z,
        "queuedTxHash": "0xf5a2b0d62e48e725697ebfcde5613744f0bd22b3506b1bea1819037982524e6c",
        "startBlock": 17757272,
        "state": "Pending",
        "totalVotesWei": "2.35266819114955989350637682e+26",
      },
      "reason": undefined,
      "support": "FOR",
      "updatedAt": 2023-07-24T11:30:47.000Z,
      "votesWei": "0",
    },
    Object {
      "address": "0xbabbf5261afbdb1f1353b574f5cae76ae916d315",
      "blockNumber": 16899221,
      "blockTimestamp": 1679683451,
      "createdAt": 2023-03-24T18:44:11.000Z,
      "id": "0x00db9beebc743dbcdcd34c1ac647bd4bb8f2c2ee3506d6f338921f1cba0c91d38c000000",
      "proposal": Object {
        "actions": Array [
          Object {
            "callData": "0x",
            "signature": "_acceptAdmin()",
            "target": "0x7d61ed92a6778f5abf5c94085739f1edabec2800",
            "value": "0",
          },
          Object {
            "callData": "0x",
            "signature": "_acceptAdmin()",
            "target": "0x1961ad247b47f4f2242e55a0e5578c6cf01f8d12",
            "value": "0",
          },
          Object {
            "callData": "0x",
            "signature": "_acceptAdmin()",
            "target": "0x714bd93ab6ab2f0bcfd2aeaf46a46719991d0d79",
            "value": "0",
          },
          Object {
            "callData": "0x",
            "signature": "_acceptAdmin()",
            "target": "0x1933f1183c421d44d531ed40a5d2445f6a91646d",
            "value": "0",
          },
          Object {
            "callData": "0x",
            "signature": "_acceptAdmin()",
            "target": "0xbced4e924f28f43a24ceedec69ee21ed4d04d2dd",
            "value": "0",
          },
          Object {
            "callData": "0x",
            "signature": "_acceptAdmin()",
            "target": "0x8f35113cfaba700ed7a907d92b114b44421e412a",
            "value": "0",
          },
          Object {
            "callData": "0x",
            "signature": "_acceptAdmin()",
            "target": "0x830dacd5d0a62afa92c9bc6878461e9cd317b085",
            "value": "0",
          },
          Object {
            "callData": "0x",
            "signature": "_acceptAdmin()",
            "target": "0xfee4428b7f403499c50a6da947916b71d33142dc",
            "value": "0",
          },
          Object {
            "callData": "0x",
            "signature": "_acceptAdmin()",
            "target": "0x7497e945f23192d1dc3095409837f67927170f2d",
            "value": "0",
          },
          Object {
            "callData": "0x",
            "signature": "_acceptAdmin()",
            "target": "0xc0534bb221c0963b213f417585e3ca581993793a",
            "value": "0",
          },
          Object {
            "callData": "0x",
            "signature": "_acceptAdmin()",
            "target": "0xf52459e8bc708d55400c167e0d77ed49641226d3",
            "value": "0",
          },
          Object {
            "callData": "0x",
            "signature": "_acceptAdmin()",
            "target": "0x307269cf921213bd2e16b22dcd4401aec21d00bf",
            "value": "0",
          },
          Object {
            "callData": "0x",
            "signature": "_acceptAdmin()",
            "target": "0x11b258a1c5ec962f43ab523be9b8fa317cb2eb69",
            "value": "0",
          },
          Object {
            "callData": "0x",
            "signature": "_acceptAdmin()",
            "target": "0x4476175752a4947de953e26b40fc8c1e145fea82",
            "value": "0",
          },
          Object {
            "callData": "0x",
            "signature": "_acceptAdmin()",
            "target": "0x0c19d213e9f2a5cbaa4ec6e8eac55a22276b0641",
            "value": "0",
          },
          Object {
            "callData": "0x",
            "signature": "_acceptAdmin()",
            "target": "0x3f2dd7adbcd464d5b0a766f7eb367bd20a249691",
            "value": "0",
          },
          Object {
            "callData": "0x",
            "signature": "_acceptAdmin()",
            "target": "0x7ab16ed27ab8b22cb625db55c0e58521969cb091",
            "value": "0",
          },
        ],
        "againstVotesWei": "0",
        "cancelDate": undefined,
        "cancelTxHash": undefined,
        "createdDate": 2023-03-24T12:13:11.000Z,
        "createdTxHash": "0xdfebb830682e5a776850d057eb908636558c0d49d8a6c1020010a76cf76db5cd",
        "description": Object {
          "description": "Abstract: After 3 weeks of testing and live network activity, the Onyx Liquidity Protocol (OLP) will now be fully shifted to the Onyx Governance and Onyx DAO . This proposal is deployed in two parts which will migrate the ownership of the OLP smart contracts to the time lock which is controlled by the Onyx Governance Onyxcoin $XCN stakers who secure and govern the network. Alongside this proposal, the OLP is now fully available via IPFS, a decentralized hosting making Onyx and OLP fully decentralized from the smart contract level to hosting.
XCN holders are asked to vote in favor for this proposal to transfer OLP to Onyx DAO and Governance.",
          "title": "OIP-9: Launch OLP Governance",
          "version": "v2",
        },
        "endBlock": 16910087,
        "executedDate": 2023-03-28T17:37:23.000Z,
        "executedTxHash": "0xddbe09d6597b72d88f93f9f31ec630c01a1c384cae9ab454a562f113ba834e0c",
        "forVotesWei": "2.873025810619257470619105486e+27",
        "id": 8,
        "isEnded": false,
        "isStarted": false,
        "proposer": "0xa8738f54c6c31ea559565a468056df233e6f4c12",
        "queuedDate": 2023-03-26T16:37:47.000Z,
        "queuedTxHash": "0x7d1960521610acfe101e477ca7a39ae8b2296f97097373347c8f196f050657a8",
        "startBlock": 16897287,
        "state": "Pending",
        "totalVotesWei": "2.873025810619257470619105486e+27",
      },
      "reason": undefined,
      "support": "FOR",
      "updatedAt": 2023-03-24T18:44:11.000Z,
      "votesWei": "0",
    },
    Object {
      "address": "0x7cb436777bcff797f465a96f6b34383e10a43483",
      "blockNumber": 16743192,
      "blockTimestamp": 1677788351,
      "createdAt": 2023-03-02T20:19:11.000Z,
      "id": "0x040e1a4716ce3f28c60835d4f6cdd8c3b6ca3e9d42e08b0d58792519aefe063942000000",
      "proposal": Object {
        "actions": Array [
          Object {
            "callData": "0x00000000000000000000000000000000000000000000002a5a058fc295ed0000",
            "signature": "setRewardPerblock(uint256)",
            "target": "0x23445c63feef8d85956dc0f19ade87606d0e19a9",
            "value": "0",
          },
        ],
        "againstVotesWei": "0",
        "cancelDate": undefined,
        "cancelTxHash": undefined,
        "createdDate": 2023-03-02T20:09:47.000Z,
        "createdTxHash": "0x40e386129ea53babe8084dc68449cb711cdb7661c2b8344ca1d15ff0d48c6f20",
        "description": Object {
          "description": "## Increase Staking Rewards

Onyx DAO proposes to the community to increase the staking vault rewards which would make the APR approximately 33% as opposed to todays approximately 6.7% staking rate.

By making this change the staking rewards still remain under the constitution threshold and creates incentive rewards for those who are securing and voting on the Onyx Protocol and Onyx DAO.

Discussion: https://community.onyx.org/t/oip-8-increase-staking-rewards/35",
          "title": "OIP-8: Increase Staking Rewards",
          "version": "v2",
        },
        "endBlock": 16755946,
        "executedDate": 2023-03-07T20:39:35.000Z,
        "executedTxHash": "0x857d533ae969d5e26fea14447bf8d5f0c81bbef97daaee4078148e626dfea6f8",
        "forVotesWei": "3.02286978509899863154493852e+26",
        "id": 7,
        "isEnded": false,
        "isStarted": false,
        "proposer": "0x09a74d465c49fd07d6eaf2028c7cff31c93d3a74",
        "queuedDate": 2023-03-05T16:31:35.000Z,
        "queuedTxHash": "0xb804ff32c580407f7d45a443de08029a6bf05670c480d324d2bca5c6e4761a7a",
        "startBlock": 16743146,
        "state": "Pending",
        "totalVotesWei": "3.02286978509899863154493852e+26",
      },
      "reason": undefined,
      "support": "FOR",
      "updatedAt": 2023-03-02T20:19:11.000Z,
      "votesWei": "2.04949456e+24",
    },
  ],
}
`;
